### PR TITLE
feat: Adding option on bottom sheet to archive use cases

### PR DESCRIPTION
**Pull request by gongracr**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We need an option to be able to archive conversations visually and consume `updateConversationArchivedStatusUseCase`.

### Solutions

Added the option on the different bottom sheets displayed when interacting with conversations options in `HomeScreen`, `GroupConversationDetailsScreen`, and `OtherUserProfileScreen`. I also added the respective snackbar messages on success and failed states.

### Dependencies (Optional)

- https://github.com/wireapp/kalium/pull/2082

Needs releases with:

- [x] GitHub link to other pull request

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

Load conversations in `HomeScreen`, long tap on a given one and click on `Move to archive` option. Check the conversation disappears and doesn't get displayed anymore. This behavior can also be replicated on Other users profile screen and on Group details screen.

### Attachments (Optional)

https://github.com/wireapp/wire-android-reloaded/assets/2468164/16095fb5-c9c7-4479-bdca-58d245b1101b

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [919](https://10.10.124.134/job/AR-build-pipeline/919/) **failed**.

**Comment by AndroidBob** [comment]
Build [928](https://10.10.124.134/job/AR-build-pipeline/928/) **failed**.

**Comment by AndroidBob** [comment]
Build [931](https://10.10.124.134/job/AR-build-pipeline/931/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/details/GroupConversationDetailsScreen.kt [hunk 1]
```diff
@@ -358,7 +358,13 @@ private fun GroupConversationDetailsContent(
                 },
                 addConversationToFavourites = bottomSheetEventsHandler::onAddConversationToFavourites,
                 moveConversationToFolder = bottomSheetEventsHandler::onMoveConversationToFolder,
-                moveConversationToArchive = bottomSheetEventsHandler::onMoveConversationToArchive,
+                moveConversationToArchive = {
+                    bottomSheetEventsHandler.onMoveConversationToArchive(
+                        conversationId = conversationSheetContent!!.conversationId,
```
Comment (Garzas) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/details/GroupConversationDetailsScreen.kt]: Maybe surround it with ?.let {} to not have a crash in some race condition 🤔

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt [hunk 2]
```diff
@@ -148,7 +151,7 @@ class ConversationListViewModel @Inject constructor(
         }
         viewModelScope.launch {
             searchQueryFlow.combine(
-                observeConversationListDetails()
+                observeConversationListDetails(fetchArchivedConversations = false)
```
Comment (Garzas) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt]: maybe rename to includeArchived?

**Review by Garzas — CHANGES_REQUESTED** [review]
Looking good 👍  I would just add some tests for onMoveConversationToArchive

**Review by Garzas — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [951](https://10.10.124.134/job/AR-build-pipeline/951/) **failed**.

**Comment by AndroidBob** [comment]
Build [965](https://10.10.124.134/job/AR-build-pipeline/965/) **failed**.

**Comment by AndroidBob** [comment]
Build [972](https://10.10.124.134/job/AR-build-pipeline/972/) **failed**.

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [985](https://10.10.124.134/job/AR-build-pipeline/985/) **failed**.

**Comment by AndroidBob** [comment]
Build [990](https://10.10.124.134/job/AR-build-pipeline/990/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7676111)(7676111)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7676111)(7676111)-staging-compat.apk)

**Pull request merged**